### PR TITLE
feat(): update commit status on github

### PR DESCRIPTION
**Pull request by Izak88**

### CODE REVIEW

File: src/api/github-commit-status.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+export function setGitHubStatusFailure(gitUrl: string, abstruseUrl: string): Promise<any> {
+  let data = {
+    'state': 'success',
+    'target_url': abstruseUrl,
+    'description': 'The abstruse build succeeded',
+    'context': 'continuous-integration/abstruse'
+  };
+
+  let header = {
+    'Authorization': 'token 11c9bd8f091ad4f7dffbf333d829133a8ff59f64',
```
Comment (jkuri) [inline src/api/github-commit-status.ts]: is this hardcoded?

File: src/api/github-commit-status.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      'description': 'The abstruse build succeeded',
+      'context': 'continuous-integration/abstruse'
+    };
+
+    let header = {
+      'Authorization': `token ${token}`,
+      'User-Agent': 'Abstruse'
+    };
+
+return sendRequest(gitUrl, data, header);
```
Comment (jkuri) [inline src/api/github-commit-status.ts]: please fix indentation.

File: src/api/github-commit-status.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    return sendRequest(gitUrl, data, header);
+}
+
+export function setGitHubStatusFailure(
+  gitUrl: string, abstruseUrl: string, token: string): Promise<any> {
+    let data = {
+      'state': 'failure',
+      'target_url': abstruseUrl,
+      'description': 'The abstruse build succeeded',
```
Comment (jkuri) [inline src/api/github-commit-status.ts]: can you fix text by replacing `abstruse` text with `Abstruse CI` please?

File: src/api/process-manager.ts [hunk 4]
```diff
@@ -131,7 +133,13 @@ export function startBuild(data: any): Promise<any> {
               data.build_id = build.id;
               delete data.repositories_id;
               delete data.pr;
-              insertBuildRun(data);
+              insertBuildRun(data)
+                .then(() => {
+                  let gitUrl =
+                    `https://api.github.com/repos/${data.head_full_name}/statuses/${data.sha}`;
```
Comment (jkuri) [inline src/api/process-manager.ts]: can we set this GitHub URL to not be hardcoded like it is now. If you are using GitHub Enterprise on your own servers the URL is different. I think that should not be hardcoded. 

I will investigate on that and make a PR after this one is merged.

Comment (Izak88) [inline src/api/process-manager.ts]: I can store that in config, or did you have something else in mind?

**Pull request merged**